File: app/src/main/java/com/glucocontrol/domain/usecase/query/GetDailyReadingsUseCase.kt
package com.glucocontrol.domain.usecase.query

import com.glucocontrol.domain.model.GlucoseReading
import com.glucocontrol.domain.repository.GlucoseRepository
import kotlinx.coroutines.flow.Flow
import java.time.LocalDate
import javax.inject.Inject

/**
 * Devuelve en tiempo real todas las lecturas de un día concreto.
 *
 * Usado por la pantalla principal (HomeScreen) para mostrar el resumen diario.
 * Las lecturas llegan ordenadas por hora ascendente, con las que no tienen hora al final.
 */
class GetDailyReadingsUseCase
@Inject
constructor(private val repository: GlucoseRepository) {
    operator fun invoke(date: LocalDate): Flow<List<GlucoseReading>> = repository.getReadingsByDate(date)
}

File: lib/domain/usecase/query/get_daily_readings_use_case.dart
import 'package:glucocontrol/domain/model/glucose_reading.dart';
import 'package:glucocontrol/domain/repository/glucose_repository.dart';

class GetDailyReadingsUseCase {
  final GlucoseRepository _repo;
  GetDailyReadingsUseCase(this._repo);

  Stream<List<GlucoseReading>> call(DateTime date) =>
      _repo.getReadingsByDate(date);
}
